docs/hbi-migration-runbook.md

```diff
@@ -0,0 +1,105 @@
+This document describes how to test the HBI migration in the ephemeral environment.
+
+## Prerequisites
+
+- `oc` is installed
+- `jq` is installed
+- `psql` is installed
+- You have access to the ephemeral environment.
+- You have cloned the [insights service deployer repository](https://github.com/project-kessel/insights-service-deployer)
+- You have cloned the [db-generator repository](https://github.com/tonytheleg/db-generator) - this repo is private and **requires access**
+    - This repo generates host and outbox HBI records for insertion into the ephemeral environment database
+
+## Steps
+
+### Base Environment Setup
+
+Login to the ephemeral environment.
+```
+oc login --token=<token> --server=<ephemeral-environment-api-url>
+```
+
+Via `insights-service-deployer`, deploy the components to the ephemeral environment.
+```
+./deploy.sh deploy
+```
+
+Via `insights-service-deployer`, deploy the HBI replication tables and kafka infra to the ephemeral environment.
+```
+# Add tables to HBI db
+./deploy.sh host-replication-tables
+```
+```
+# Add kessel kafka infra
+./deploy.sh host-replication-kafka
+```
+
+### Generate & Insert HBI records
+If you don't have host data already generated, generate it from the `db-generator` repository.
+
+The following generates 10 host files with 1000 host records each (hosts are counted in hundreds and multiplied by the number of files). This means the following command will generate 10,000 host records total.
+```
+go run main.go --num-hosts 10 --num-files 10 --type hosts 
+```
+
+Grab the host-inventory-db database `user` secret from the ephemeral environment.
+```
+oc get secrets host-inventory-db -o json | jq '.data["db.user"]' -r | base64 -d
+```
+
+Now port forward to the host-inventory-db database
+```
+oc port-forward service/host-inventory-db 5433:5432
+```
+
+Next, insert the HBI records into the ephemeral environment database.
+```
+psql -h localhost -p 5433 -U <user> -d host-inventory-db -f <path-to-generated-host-file>
+```
+
+You can validate the records are inserted by checking the `host-inventory-db` database:
+```
+psql -h localhost -p 5433 -U <user> -d host-inventory-db -c "select count(*) from hosts" -x
+```
+
+### Start the Migration
+
+To start the HBI migration, you just need to set the host migration connector to `running`
+
+```
+oc patch kafkaconnector "hbi-migration-connector" --type='merge' -p='{"spec":{"state":"running"}}'
+```
+
+### Validating the Migration
```

Comment:
Maybe overkill, but not sure if we want to provide steps on validating messages via kafka

```shell
oc rsh kessel-kafka-connect-connect-0
./bin/kafka-console/consumer.sh --bootstrap-server $KAFKA_CONNECT_BOOTSTRAP_SERVERS --topic host-inventory.hbi.hosts --property print.key=true --property print.headers=true --from-beginning

./bin/kafka-console/consumer.sh --bootstrap-server $KAFKA_CONNECT_BOOTSTRAP_SERVERS --topic outbox.event.hbi.hosts --property print.key=true --property print.headers=true --from-beginning
```

Note: `KAFKA_CONNECT_BOOTSTRAP_SERVERS` is already set inside the connect pod, no need to grab anything